torchvision/transforms/functional_tensor.py

```diff
@@ -48,3 +48,74 @@ def crop(img, top, left, height, width):
         raise TypeError('tensor is not a torch image.')
 
     return img[..., top:top + height, left:left + width]
+
+
+def adjust_brightness(img, brightness_factor):
+    """Adjust brightness of an RGB image.
+
+    Args:
+        img (Tensor): Image to be adjusted.
+        brightness_factor (float):  How much to adjust the brightness. Can be
+            any non negative number. 0 gives a black image, 1 gives the
+            original image while 2 increases the brightness by a factor of 2.
+
+    Returns:
+        Tensor: Brightness adjusted image.
+    """
+    if not F._is_tensor_image(img):
+        raise TypeError('tensor is not a torch image.')
+
+    black_img = torch.zeros(img.shape, device=img.device)
+
+    return _blend(img, black_img, brightness_factor)
+
+
+def adjust_contrast(img, contrast_factor):
+    """Adjust contrast of an RGB image.
+
+    Args:
+        img (Tensor): Image to be adjusted.
+        contrast_factor (float): How much to adjust the contrast. Can be any
+            non negative number. 0 gives a solid gray image, 1 gives the
+            original image while 2 increases the contrast by a factor of 2.
+
+    Returns:
+        Tensor: Contrast adjusted image.
+    """
+    if not F._is_tensor_image(img):
+        raise TypeError('tensor is not a torch image.')
+
+    mean = torch.mean(_rgb_to_grayscale(img))
+    mean_img = mean * torch.ones(img.shape, device=img.device)
+
+    return _blend(img, mean_img, contrast_factor)
+
+
+def adjust_saturation(img, saturation_factor):
+    """Adjust color saturation of an RGB image.
+
+    Args:
+        img (Tensor): Image to be adjusted.
+        saturation_factor (float):  How much to adjust the saturation. 0 will
+            give a black and white image, 1 will give the original image while
+            2 will enhance the saturation by a factor of 2.
+
+    Returns:
+        Tensor: Saturation adjusted image.
+    """
+    if not F._is_tensor_image(img):
+        raise TypeError('tensor is not a torch image.')
+
+    num_channels = img.shape[0]
+    gray_img = _rgb_to_grayscale(img).repeat(num_channels, 1, 1)
```

Comment:
This works nicely, but a very nice thing with PyTorch tensors is that we have automatic broadcasting on the left-most dimensions.
So if we just pass `_rgb_to_grayscale(img)` to `_blend`, this should work I believe, because the extra dimension will be automatically added and expanded. This has the benefit to save some memory as well, because `.repeat` allocates new memory.

Reply:
Done.